guides/hack/11-built-in-types/28-shapes.md

```diff
@@ -95,6 +95,19 @@ function takes_server(Server $s) {
 }
 ```
 
+While you can't instantiate these types directly, you can create a my_function_arguments_t by simply defining a shape that has all of the required fields. The typechecker will automatically infer if your shape can be accepted by code() or not.
+
+``` Hack
```

Comment:
no space; also please provide a filename, like:

~~~
```values-of-type.hack
~~~

Then, when built, a typechecker output file will be generated; that should ideally be committed, then if something changes with shapes, the build process will detect that the example is no longer valid.

Reply:
This would also need a filename to be specified for the example that defines the shape, or that definition will be undefined

Reply:
oo this is really cool!

Reply:
done, and committed!